extension/extensionauth/client.go

```diff
@@ -42,25 +49,29 @@ type ClientRoundTripperFunc func(base http.RoundTripper) (http.RoundTripper, err
 // ClientPerRPCCredentialsFunc specifies the function that returns a PerRPCCredentials that can be used to authenticate gRPC requests.
 type ClientPerRPCCredentialsFunc func() (credentials.PerRPCCredentials, error)
 
-var _ Client = (*defaultClient)(nil)
+var (
+	_ HTTPClient = (*Client)(nil)
+	_ GRPCClient = (*Client)(nil)
+)
 
-type defaultClient struct {
+// Client is an Extension that can be used as an HTTP and gRPC authenticator for the configauth.Authentication option.
+type Client struct {
```

Comment:
I don't think we need this anymore if we go with the optional interface as suggested.